### PR TITLE
[Refactor](opt) Opt rf and remove unless code

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

File: be/src/exprs/runtime_filter.cpp [hunk 1]
```diff
@@ -735,7 +735,9 @@ class RuntimePredicateWrapper {
         _is_bloomfilter = true;
         // we won't use this class to insert or find any data
         // so any type is ok
-        _context.bloom_filter_func.reset(create_bloom_filter(PrimitiveType::TYPE_INT));
+        _context.bloom_filter_func.reset(create_bloom_filter(_column_return_type == INVALID_TYPE
+                                                                     ? PrimitiveType::TYPE_INT
```
Comment (yiguolei) [inline be/src/exprs/runtime_filter.cpp:739]: why assign type int when it is invalid type?

File: be/src/exprs/runtime_filter.h [hunk 2]
```diff
@@ -276,20 +273,21 @@ class IRuntimeFilter {
...
                                  const MergeRuntimeFilterParams* param, ObjectPool* pool,
                                  std::unique_ptr<RuntimePredicateWrapper>* wrapper);
     static Status create_wrapper(RuntimeFilterParamsContext* state,
                                  const UpdateRuntimeFilterParams* param, ObjectPool* pool,
                                  std::unique_ptr<RuntimePredicateWrapper>* wrapper);
+
     static Status create_wrapper(RuntimeFilterParamsContext* state,
-                                 const UpdateRuntimeFilterParamsV2* param, ObjectPool* pool,
-                                 std::unique_ptr<RuntimePredicateWrapper>* wrapper);
+                                 const UpdateRuntimeFilterParamsV2* param,
```
Comment (yiguolei) [inline be/src/exprs/runtime_filter.h:284]: The lifecycle of wrapper is not consistency. Sometimes it is in pool, but sometimes we use unique ptr

File: be/src/exprs/runtime_filter.cpp (hunk 1 above)
Comment (HappenLee) [reply be/src/exprs/runtime_filter.cpp:739]: the up comment tell the reason: 
```we won't use this class to insert or find any data
   so any type is ok
```

after change the logic, the `_column_return_type` be valid in `apply filter`, can reuse the mem of it

File: be/src/exprs/runtime_filter.h (hunk 2 above)
Comment (HappenLee) [reply be/src/exprs/runtime_filter.h:284]: This lifecycle is more complex in RF, the use of Pool is to better reuse memory, and the next PR will refactor this part of the code and unify it into Pool.
after update the patch，next version of doris only use V2， we can delete the old `update filter` api

**Review by yiguolei — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**